specifyweb/frontend/js_src/lib/components/SearchDialog/SelectRecordSet.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+
+import { useAsyncState } from '../../hooks/useAsyncState';
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { commonText } from '../../localization/common';
+import type { RA } from '../../utils/types';
+import { Ul } from '../Atoms';
+import { Button } from '../Atoms/Button';
+import { Input, Label } from '../Atoms/Form';
+import { fetchCollection } from '../DataModel/collection';
+import type { AnySchema } from '../DataModel/helperTypes';
+import { schema } from '../DataModel/schema';
+import type { Collection, SpecifyTable } from '../DataModel/specifyTable';
+import { getTableById } from '../DataModel/tables';
+import { softError } from '../Errors/assert';
+import { userInformation } from '../InitialContext/userInformation';
+import { Dialog, LoadingScreen } from '../Molecules/Dialog';
+import { usePaginator } from '../Molecules/Paginator';
+import { TableIcon } from '../Molecules/TableIcon';
+
+export function SelectRecordSets<SCHEMA extends AnySchema>({
```

Comment:
Here is a version using RS dialog: https://github.com/specify/specify7/pull/5263/commits/1745d06aea57905709b7c35dbcdefda7b1e1565b
I personally prefer the one with the new SelectRecordSet component. 
Using the original RS dialog adds complexity into that component just to get rid of one small piece of code that was reused <const [recordSets] = useAsync...>

Reply:
ps: 
Still need to add conditionals to the RS dialog to only display the checkbox when needed and conditional for the table className for the number of columns

Reply:
regarding https://github.com/specify/specify7/commit/1745d06aea57905709b7c35dbcdefda7b1e1565b, complexity can be reduced by, instead of adding features to the RecordSetsDialog component, creating a component on top of RecordSetsDialog that contains the additions you need

For example, the entire isAddResourceToParent and handleAddResource logic can be turned into two 2 new props in RecordSetsDialog like `getChild?: (row: RecordSet) => JSX.Element` and `extraButtons?: JSX.Element`) - that  adds hardly any new complexity to RecordSetsDialog, while letting you reuse all the common logic in the uber-component

but I am ok if you considered this solution and decided not to pursue it